### PR TITLE
upload chapter 10's translation

### PR DESCRIPTION
**Pull request by Zheng7426**
Hi @yukie1412, please review it when you have time. Thanks.

### CODE REVIEW

File: 10_modules.md [hunk 1]
```diff
@@ -14,181 +14,96 @@ quote}}
```
Comment (ghost) [inline 10_modules.md:14]: alt should also be localized imo, for a11y

File: 10_modules.md [hunk 2]
```diff
@@ -14,181 +14,96 @@ quote}}
...
+拥有可供下载的高质量的包是相当有价值的。这意味着我们通常可以避免重新发明一个已经被 100 个人写过的程序，而只需要按几个键就可以得到一个可靠的、几经测试的程序的实现。
 
 {{index maintenance}}
 
-Software is cheap to copy, so once someone has written it,
-distributing it to other people is an efficient process. But writing
-it in the first place _is_ work, and responding to people who have
-found problems in the code, or who want to propose new features, is
-even more work.
+复制软件是廉价的，所以一旦有人编写了该软件，将其分发给其他人的过程是有效率的。但是一开始写它的时候 _确实是_ 不容易的工作，而去回应那些在代码中发现问题、或想提出新增特性的人，则意味着更多的工作。
```
Comment (ghost) [inline 10_modules.md]: i'm not sure about this sentence. original sentence doesn't have "不容易"

File: 10_modules.md [hunk 3]
```diff
@@ -201,37 +116,26 @@ const weekDay = function() {
...
 
-If we want to make dependency relations part of the code, we'll have
-to take control of loading dependencies. Doing that requires being
-able to execute strings as code. JavaScript can do this.
+如果我们想让依赖关系成为代码的一部分，我们会需要对依赖项的加载进行控制。做到这一点需要将字符串作为代码来执行，JavaScript 足以胜任。 
 
 {{id eval}}
 
-## Evaluating data as code
+## 视数据为代码
```
Comment (yukie1412) [inline 10_modules.md:130]: // TODO

**Review by ghost — COMMENTED** [review]
apologies for this late review

File: 10_modules.md (hunk 2 above)
Comment (Zheng7426) [reply 10_modules.md]: Good point. I will delete "不容易".

File: 10_modules.md [hunk 4]
```diff
@@ -241,64 +145,45 @@ function evalAndReturnX(code) {
...
+因为加载器将模块代码包装在一个函数里，这些模块会自动获取它们自己的本地作用域。它们只需要调用 `require` 来访问依赖项，并且把它们的接口放在与 `exports` 绑定的对象中。
 
 {{index "formatDate module", "Date class", "ordinal package", "date-names package"}}
 
-This example module provides a date-formatting function. It uses two
-((package))s from NPM—`ordinal` to convert numbers to strings like
-`"1st"` and `"2nd"`, and `date-names` to get the English names for
-weekdays and months. It exports a single function, `formatDate`, which
-takes a `Date` object and a ((template)) string.
+以下的示例模块提供了一个格式化日期的函数。它用了两个来自 NPM 的((包)) —— 将数字转换成类似 `"1st"` 和
```
Comment (ghost) [inline 10_modules.md]: nit: unnecessary newline

File: 10_modules.md [hunk 5]
```diff
@@ -318,27 +203,23 @@ exports.formatDate = function(date, format) {
...
-that depend on it get access to it. We could use the module like this:
+该模块把它的接口函数添加到 `exports`，从而让依赖它的模块可以访问它。我们可以这样使用该模块：
 
 ```
 const {formatDate} = require("./format-date");
 
 console.log(formatDate(new Date(2017, 9, 13),
                        "dddd the Do"));
-// → Friday the 13th
+// → Friday the 13th    周五，本月13号
```
Comment (ghost) [inline 10_modules.md]: it's not "this month"

File: 10_modules.md [hunk 6]
```diff
@@ -447,204 +292,97 @@ export function formatDate(date, format) { /* ... */ }
...
 
-Many projects are written using ES modules and then automatically
-converted to some other format when published. We are in a
-transitional period in which two different module systems are used
-side by side, and it is useful to be able to read and write code in
-either of them.
+很多项目使用 ES 模块编写，之后在发布时被自动转化成其他的格式。我们正处于这两种模块系统被并排使用的过渡时期，在这两者的任何一种模块系统下都能够读写代码是大有裨益的。
 
-## Building and bundling
+## 开发及打包
```
Comment (yukie1412) [inline 10_modules.md:327]: // TODO

File: 10_modules.md [hunk 7]
```diff
@@ -658,41 +395,28 @@ for (let node of Object.keys(roadGraph)) {
 }
 
 console.log(find_path(graph, "Post Office", "Cabin"));
-// → ["Post Office", "Alice's House", "Cabin"]
+// → ["Post Office", "Alice's House", "Cabin"]    邮局，爱丽丝的家， 小屋
```
Comment (ghost) [inline 10_modules.md:398]: unnecessary extra space

**Review by ghost — APPROVED** [review]

**Pull request merged**